include/RAJA/pattern/params/reducer.hpp

```diff
@@ -83,16 +55,38 @@ namespace detail
   // Basic Reducer
   //
   //
-  template <typename Op, typename T>
+
+  // Basic data type Reducer
+  // T must be a basic data type
+  // IndexType not used here
+  // VType must be ValOp<T, Op>
+  template <template <typename, typename, typename> class Op, typename T, typename IndexType, typename VType>
```

Comment:
Why does this have IndexType?

Reply:
If we remove `IndexType` here, then the `Reducer` objects in `init`, `combine`, and `resolve` will require duplication. As of now, we can just keep those calls generic, able to be used for the non-index and index cases. I think keeping `IndexType` here in a non-index `Reducer` is a very small price to pay for this convenience. I can't find a way to quote my own PR, but you can see this in something like `policy/sequential/params/reduce.hpp`.

Reply:
IndexType is known by T in the case where it is used so it doesn't have to be a separate template param.